services/github/github-deployments.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      deployments: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestStatus: Joi.object({
+                state: Joi.string(),
```

Comment:
We should use a stricter set of validations here based on the known/expected status values

Reply:
For reference, here's the raw values from GitHub that Shields will see (note the all upper case) https://developer.github.com/v4/enum/deploymentstatusstate/